### ISSUE TITLE
addGitLabMRComment, update existing comment on subsequent commits

### ISSUE THREAD

**Issue by anton-johansson** [jenkinsci/gitlab-plugin#575]
## Issue

It would be nice if subsequent comments in the same merge request would cause the Jenkins plug-in to update the existing comment instead of adding new ones. Of course, controlled with a setting.

### Context
- **Gitlab plugin version**: 1.4.5
- **Gitlab version**: 8.12
- **Jenkins version**: 2.46.3
- **Job type**: Pipeline

### Problem description

I've not yet integrated this plug-in in our build cycle, but during my tests I noticed that merge requests with a lot of commits would get quite spammed by Jenkins. It would be great to be able to update the existing comment, IMO! I'm not sure what the possibilities exist in the GitLab API to do this, though.

**Comment by omehegan** [comment]
Yes, this is one reason I don't really like the "comment on build result" pattern. Especially ever since GitLab has been able to show build pass/fail status with its built-in UI. But we can consider this if someone wants to submit a PR.

**Comment by anton-johansson** [comment]
I want the "comment on build" even when I use "update build status", because I can provide a detailed summary of the build, such as unit test failures, code coverage and code style analysis reports.

As far as I can see, we can use this method:
`GET /projects/:id/merge_requests/:merge_request_iid/notes`

... to get all existing comments. Then see if there's anyone commented by Jenkins. Then use this method to update it:
`PUT /projects/:id/merge_requests/:merge_request_iid/notes/:note_id`

However, since comments are created using an access token, it's hard to find the comment that is commented by Jenkins... It would be awesome if GitLab provided a way of getting comments of a MR posted by the current access token, but I can't find anything like that. :(

**Comment by anton-johansson** [comment]
I have an open issue in GitLab:
https://gitlab.com/gitlab-org/gitlab-ce/issues/34738

If this gets solved and we have an API that we can use in a good way here, I'll give a PR a go!
